### ISSUE TITLE
Impacted by Google deprecating OAuth out-of-band (oob) flow on Oct 3, 2022

### ISSUE THREAD

**Issue by charmoney** [munir131/attachment-downloader#28]
I received a Google email with the subject line "Migrate your OAuth out-of-band flow to an alternative method before Oct. 3, 2022" on May 3, 2022. It identified to OAuth credentials created for this tool as being impacted.

https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html#disallowed-oob

Please consider updating the tool to use an allowed OAuth mechanism.

**Comment by munir131** [comment]
@charmoney Thanks for informing that. I will updated toll to use an allowed OAuth mechanism.

**Comment by charmoney** [comment]
Just wanted to check in on this as the 10/3/2022 deadline approaches. Does one of the dependency updates from eaf2c7994aba00799e15179397071a3369259f56 support other oauth mechanisms?

**Comment by munir131** [comment]
@charmoney I haven't get time to verify that. I don't think dependency updates will support other oauth mechanisms.

**Comment by munir131** [comment]
@charmoney I just created new application on google console and it is as per supported oauth mechanisms. So, you need to create new app from console. As per [doc](https://developers.googleblog.com/2022/02/making-oauth-flows-safer.html#instructions-oob)  redirect url should contain `oob` 
Old app json file content:
```"redirect_uris":["urn:ietf:wg:oauth:2.0:oob","http://localhost"]```
New app json file content:
```"redirect_uris":["http://localhost"]```

**Comment by charmoney** [comment]
I pulled updates to this tool, re-installed all node dependencies, created a new Desktop app, and set up the `credentials.json`. Just as you showed, the new json file contains `"redirect_uris":["http://localhost"]` without the oob.

I am able to complete the Google side of the authorization but then get sent back to `http://localhost/?code=...` with an ERR_CONNECTION_REFUSED in the browser. Is the node application expected to be running a temporary localhost server?

I was able to copy the code out of the failed URL, paste it into the terminal, and authorization completed successfully.

Thanks!
